test/libsolidity/SemanticTest.cpp

```diff
@@ -312,6 +338,14 @@ TestCase::TestResult SemanticTest::runTest(
 			test.setRawBytes(move(output));
 			test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName(m_sources.mainSourceFile)));
 		}
+
+		vector<string> effects;
+		for (auto const& hook: m_sideEffectHooks)
+			for (auto const& effect: hook(test.call()))
+				effects.emplace_back(effect);
```

Comment:
The suggestion from https://github.com/ethereum/solidity/pull/11048#discussion_r624207467 can now be applied here:

```suggestion
		for (SideEffectHook const& hook: m_sideEffectHooks)
			effects += hook(test.call())
```

Also, not sure if I'd call it better but with ranges-v3 something like this might also work:
```c++
auto callHook = [&test](SideEffectHook const& _hook){ return _hook(test.call()); };
test.setSideEffects(m_sideEffectHooks | ranges::views::transform(callHook) | ranges::views::join() | ranges::to<vector>());
```

Reply:
I think the "traditional" implementation is easier to read.

Reply:
Yeah. That's basically how you would do a list comprehension in Python. It's not very readable in C++ though and the namespaces make it even worse :(